### ISSUE TITLE
Remove identity token from the multipart response

### ISSUE THREAD

**Issue by ndr-brt** [eclipse-edc/Connector#2345]
# Feature Request

Remove identity token from the multipart response

## Which Areas Would Be Affected?
-

## Why Is the Feature Desired?
Identity should be verified on another level

## Solution Proposal
-

**Comment by dileep-ramachandrarao** [comment]
Continuing from [previous discussion](https://github.com/eclipse-edc/Connector/issues/2039), does the ids:token (DAT from DAPS) needs to be removed from all IDS multipart response messages? In such a case, all modules under [data-protocols/ids](https://github.com/eclipse-edc/Connector/tree/main/data-protocols/ids) would be affected. 
Maybe a question beyond the scope of this issue, is the ids:token would be removed from IDS multipart request messages as well? In such a case, EDC would not support DAPS?

**Comment by jimmarino** [comment]
> Continuing from [previous discussion](https://github.com/eclipse-edc/Connector/issues/2039), does the ids:token (DAT from DAPS) needs to be removed from all IDS multipart response messages? In such a case, all modules under [data-protocols/ids](https://github.com/eclipse-edc/Connector/tree/main/data-protocols/ids) would be affected. Maybe a question beyond the scope of this issue, is the ids:token would be removed from IDS multipart request messages as well? In such a case, EDC would not support DAPS?

The question of "support for DAPS" is a separate issue. By that, I assume support for DAPS is the ability for an EDC client to request a security token from DAPS and present it to a remote endpoint to satisfy a requirement for security credentials.

Validating the security credentials of a remote endpoint after the client has made a request doesn't make much sense. The trust linkage between the remote endpoint and the client needs to be established before the request is made and depends on the identity system used (DAPS is only one implementation).

The IDS specifications are in the process of being updated to address a number of issues. Part of the changes will entail the sunsetting of IDS multipart and the current information model classes. This will result in a streamlined protocol that will be much easier to implement. We should be able to start this work in the EDC over the next month or so.

**Comment by dileep-ramachandrarao** [comment]
Thanks @jimmarino for the clarification.
If the new IDS specification would result in sun-setting of IDS multipart messages, then even the ids:token would also be removed. As of now, ids:token in response message does not have any significance as well. 
This issue would then be addressed as part of the larger issue to remove multipart message, when the new IDS specification is available. In such a case, does it make sense to remove ids:token from response message for now, knowing that the multipart message would be removed sooner?

**Comment by jimmarino** [comment]
> Thanks @jimmarino for the clarification. If the new IDS specification would result in sun-setting of IDS multipart messages, then even the ids:token would also be removed. As of now, ids:token in response message does not have any significance as well. This issue would then be addressed as part of the larger issue to remove multipart message, when the new IDS specification is available. In such a case, does it make sense to remove ids:token from response message for now, knowing that the multipart message would be removed sooner?

Yes, that's why we have this issue.

**Comment by ndr-brt** [comment]
> In such a case, does it make sense to remove ids:token from response message for now, knowing that the multipart message would be removed sooner?
Yes, because we don't know when the new protocol will be on place, and removing the token from the response would probably take some minutes or so.

**Comment by jimmarino** [comment]
> > In such a case, does it make sense to remove ids:token from response message for now, knowing that the multipart message would be removed sooner?
> 
> Yes, because we don't know when the new protocol will be on place, and removing the token from the response would probably take some minutes or so.

It may also be worth mentioning there is the potential for the returned token to be misused by a malicious actor, so best to remove that possibility.

**Comment by juliapampus** [comment]
I agree to @dileep-ramachandrarao and vote for closing this issue. But I also vote for not re-opening #2039.

@jimmarino @ndr-brt I know that we will implement the new protocol and that there is no DAT anymore. But still, we have this IDS v1 implementation and, following our decision to not surprisingly remove features but mark them as deprecated and keep them a few milestones, this old implementation will be in our repo in parallel to the v2 protocol. So people are free to use it, and will do for some while due to the natural adoption processes. 

In addition, we always claimed that we implemented v1 as "complete" as documentation existed. The specification of v1 says that the DAT is an attribute of **every** IDS multipart message, be it request or response. @jimmarino There is no need to question the sense of a DAT in v1 now, these are facts. We have been there before, and this is why we removed it in v2.

I agree to not putting effort in improving deprecated stuff, especially when this touches non-IDS implementation (in this case the Oauth2 extension) - what would be the case, right? @dileep-ramachandrarao 

However, I also am no fan of changing APIs for vague reasons. We have this v1 for a while now and know this will be deprecated, so why changing things? We do not have a wrong verification, it is "just" missing. From security perspectiv, you may rather declare issue #2039 as bugfix that is labelled as "out-of-scope".

**Comment by jimmarino** [comment]
Let's discuss this during our next committer's round. This is a bit more complex:

1. The existing token is not used, so removing it will not impact users
2. There is a potential series security issue with propagating the token to the client (we can discuss details offline since it is security related). Removing the unused token addresses this, otherwise we will need to implement a fix.
3. I'm not sure if we ever claimed to implement "all" of the "old" IDS protocol since there is no normative documentation that would allow us to make that claim and we also had to make changes to the protocol for various technical reasons (e.g. scalability).

In my view, the most important issue is point 2 and the amount of work to simply remove the token that no one is using is minimal.

**Comment by dileep-ramachandrarao** [comment]
Hello @juliapampus  and @jimmarino,
I would be interested to know the decision on this issue.
Thanks in advance!

**Comment by juliapampus** [comment]
Will be resolved with new dsp modules
